### PR TITLE
Implement logic to lazy load GridField in the frontend

### PR DESCRIPTION
**Pull request by maxime-rainville**
Fixes https://github.com/silverstripe/silverstripe-cms/issues/2276

Depends on https://github.com/silverstripe/silverstripe-framework/pull/8460

This PR updates the GridField legacy JS logic to allow GridField data to be loaded once it's parent tab is activated. It also has the side benefit of updating the loading animation of GridField to be contained to the GridField table.

# Known compatibity issue with  `colymba/gridfield-bulk-editing-tools`
There's a bug when the lazy loading is combined with `colymba/gridfield-bulk-editing-tools`. If you start on a page that doesn't have a GridField and access a page with GridField that has the lazy loading and bulk editing component enable, the results are not properly fetch. (The Comments in the CWP kitchen are a good example of this). I think it's a bug with with `gridfield-bulk-editing-tools`, so we can raise a separate issue. If it's this is a blocker we can add an exception to disable the lazy loading if we detect `gridfield-bulk-editing-tools` is present.

### CODE REVIEW

File: client/src/legacy/GridField.js [hunk 1]
```diff
@@ -41,14 +67,21 @@ $.entwine('ss', function($) {
...
+      tbody.append($('<tr />').append(loadingCell));
 
-      $.ajax($.extend({}, {
+      var request = $.ajax($.extend({}, {
         headers: {"X-Pjax" : 'CurrentField'},
         type: "POST",
         url: this.data('url'),
         dataType: 'html',
-        success: function(data) {
+        async:true,
```
Comment (robbieaverill) [inline client/src/legacy/GridField.js]: Space after `:`

File: client/src/legacy/GridField.js [hunk 2]
```diff
@@ -41,14 +67,21 @@ $.entwine('ss', function($) {
...
+        success: function (data) {
+          console.dir(data)
```
Comment (robbieaverill) [inline client/src/legacy/GridField.js]: Remove debug code

File: client/src/legacy/TabSet.js [hunk 3]
```diff
@@ -77,7 +84,23 @@ $.entwine('ss', function($){
 				if(!matches) return;
 				$(this).attr('href', document.location.href.replace(/#.*/, '') + matches[0]);
 			});
-		}
+		},
+
+    /**
+     * @func lazyLoadGridFields
+     * @desc Find all the lazy loadable gridfield in the panel and trigger their reload.
+     * @param panel
```
Comment (robbieaverill) [inline client/src/legacy/TabSet.js]: missing type

File: client/src/legacy/GridField.js (hunk 1 above)
Comment (maxime-rainville) [reply client/src/legacy/GridField.js]: `async:true` should not have been there ... that was a remnant of me knocking my head against the wall and trying stuff.

File: client/src/legacy/GridField.js (hunk 2 above)
Comment (maxime-rainville) [reply client/src/legacy/GridField.js]: Gone.

File: client/src/legacy/TabSet.js (hunk 3 above)
Comment (maxime-rainville) [reply client/src/legacy/TabSet.js]: Fixed.

**Comment by maxime-rainville** [comment]
All ready to go.

File: client/src/legacy/GridField.js [hunk 4]
```diff
@@ -19,16 +19,42 @@ $.entwine('ss', function($) {
         this.fixColumns();
         this.injectSearchButton(false);
       }
+
+      if (this.is('.grid-field-lazy--loadable') && (
```
Comment (robbieaverill) [inline client/src/legacy/GridField.js]: The class names in the framework PR are

```
            'grid-field--lazy-loadable' :
            'grid-field--lazy-loaded');
```

**Comment by robbieaverill** [comment]
@maxime-rainville re "Known compatibity issue with colymba/gridfield-bulk-editing-tools" can you please raise that issue?

**Pull request merged**